js/src/helpers/addCheckToChecklist.js

```diff
@@ -0,0 +1,83 @@
+import { __ } from "@wordpress/i18n";
+
+/**
+ * Adds a "No focus keyword was entered" result when no focus keyphrase
+ * was entered.
+ *
+ * @param {Object[]} checklist         The checklist on which to add the check.
+ * @param {boolean}  hasFocusKeyphrase Whether there is a focus keyphrase.
+ *
+ * @returns {void}
+ */
+export function addFocusKeyphraseCheck( checklist, hasFocusKeyphrase ) {
+	if ( ! hasFocusKeyphrase ) {
+		checklist.push( {
+			label: __( "No focus keyword was entered", "wordpress-seo" ),
+			score: "bad",
+		} );
+	}
+}
+
+/**
+ * Adds the readability check if readability analysis is active.
+ *
+ * @param {Object[]} checklist                   The checklist on which to add the readability check.
+ * @param {Object}   readabilityScore            The readability score.
+ * @param {boolean}  isReadabilityAnalysisActive Whether readability analysis is active or not.
+ *
+ * @returns {void}
+ */
+export function addReadabilityCheck( checklist, readabilityScore, isReadabilityAnalysisActive ) {
+	if ( isReadabilityAnalysisActive ) {
```

Comment:
In terms of WordPress this actually is a `maybeReadabilityCheck`

It is also weird to pass the decision maker as argument. I would fetch the setting in this place instead of passing its value.